pandas/_libs/indexing.pyx

```diff
@@ -20,4 +20,5 @@ cdef class _NDFrameIndexerBase:
         ndim = self._ndim
         if ndim is None:
             ndim = self._ndim = self.obj.ndim
+            assert not ndim > 2, 'Internal Error: NDFrameIndexer ndim > 2'
```

Comment:
is this ever raised in user code? what happens when a 3-d indexer is passed?

Reply:
can you add some sort of test for this (also this should raise a ValueError if its user code)

Reply:
> is this ever raised in user code? what happens when a 3-d indexer is passed?

added some tests for 3d indexing (integer ndarray only at the moment). not sure if this was what was required. tests is messy with mixture of some getting and setting not raising and those that raise produce a range of error messages.

> can you add some sort of test for this (also this should raise a ValueError if its user code)

will look into this next.

Reply:
> also this should raise a ValueError if its user code

since NDFrame is not part of the API, i'm not sure how this can be raised from user code. (other than Panel)

i've added a test for NDFrame initialized with 3d array, but it only raises this exception with `.ix`